### PR TITLE
[SYCL] Enable test for ZE_DEBUG=4

### PR DESCRIPTION
**Pull request by bso-intel**
As I fixed memory leaks in https://github.com/intel/llvm/pull/5541,
we need to enable ZE_DEBUG=4 tests for these test cases.

### CODE REVIEW

**Comment by bso-intel** [comment]
The failure in undefined-symbol.cpp is intentional.
We should never put // UNSUPPORTED: host for any test.
The reason is that GPU_PLACE_HOLER, CPU_PLACE_HOLDER, ACC_PLACE_HOLDER all include the host target by default.
So, if there is //UNSUPPORTED:host tag, this means this test does is hidden from any testing.
So, I decided to reveal this failure so that someone can fix this failure.

**Comment by bso-intel** [comment]
When I added XFAIL: gpu, CI complains it passes unexpected.
When I remove XFAIL:gpu, CI complains it fails on gpu.
So, I don't know how to fix the CI issue.
And this CI issue is not the scope of my PR.
I restored "//UNSUPPORTED: host" to make CI happy.

File: SYCL/KernelAndProgram/undefined-symbol.cpp [hunk 1]
```diff
@@ -1,5 +1,5 @@
 // XFAIL: cuda || hip
-// UNSUPPORTED: host,ze_debug-1,ze_debug4
+// UNSUPPORTED: host
```
Comment (vladimirlaz) [inline SYCL/KernelAndProgram/undefined-symbol.cpp]:
```suggestion
```

File: SYCL/KernelAndProgram/undefined-symbol.cpp [hunk 2]
```diff
@@ -1,5 +1,4 @@
 // XFAIL: cuda || hip
```
Comment (vladimirlaz) [inline SYCL/KernelAndProgram/undefined-symbol.cpp]:
```suggestion
// XFAIL: cuda || hip || (opencl && gpu)
```

**Review by vladimirlaz — APPROVED** [review]

**Pull request merged**